### PR TITLE
[#55] Update owner hook logic to match FA2

### LINKED ISSUE #55: Update owner hook logic to match FA2
# Clarification and motivation

In the latest [FA2](https://gitlab.com/tzip/tzip/-/blob/b916f32718234b7c4016f46e00327d66702511a2/proposals/tzip-12/tzip-12.md) version there are some changes regarding transfer hooks. We should update our code and specification.

# Acceptance criteria

Code and spec match latest FA2 with regards to transfer hooks. Specifically:
1. `Owner_no_op` is renamed to `Owner_no_hook`.
2. `transfer_descriptor_param_michelson` is updated, it now has a list of destinations.
3. Specification and tests are updated accordingly.

Note: since FA2 is frequently updated let's assume that latest means `b916f32718234b7c4016f46e00327d66702511a2`.

### PR DESCRIPTION
**Pull request by awkure**
## Description

In the latest FA2 version there are some changes regarding transfer hooks. We should update our code and specification. Note, that specification wasn't updated since the actual renaming appeared in #57. 

## Related issue(s)

Resolves #55

#### Related changes (conditional)

- Tests
  - [X] If I added new functionality, I added tests covering it.
  - [X] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [X] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [X] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [X] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/stablecoin/operator.ligo [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 #include "hooks.ligo"
 
+function e (const u : unit) : unit is unit
```
Comment (gromakovsky) [inline ligo/stablecoin/operator.ligo:10]: What is it?

File: docs/specification.md [hunk 2]
```diff
@@ -372,35 +368,31 @@ Parameter (in Michelson)
...
   (pair
-    (or %self
-      (unit %self_transfer_permitted)
-      (unit %self_transfer_denied)
-    )
     (pair
       (or %operator
-        (unit %operator_transfer_permitted)
-        (unit %operator_transfer_denied)
+        (unit %owner_transfer)
```
Comment (sras) [inline docs/specification.md:373]: `Owner_or_operator_transfer` is also required here?

File: ligo/stablecoin/actions.ligo [hunk 3]
```diff
@@ -336,8 +336,9 @@ function is_operator_action
   { const operator_transfer_policy : operator_transfer_policy_ =
       Layout.convert_from_right_comb ((store.permissions.operator : operator_transfer_policy))
   ; case operator_transfer_policy of
-      Operator_transfer_permitted (u) -> skip
-    | Operator_transfer_denied (u) -> failwith ("TX_DENIED")
+      No_transfer -> failwith ("TX_DENIED")
```
Comment (sras) [inline ligo/stablecoin/actions.ligo:339]: Why do we have to make this check. Aren't we hard coding the behavior?

Comment (gromakovsky) [reply ligo/stablecoin/actions.ligo:339]: I suppose this PR just needs to be rebased.

**Pull request merged**